### DISCUSSION TITLE
[Script request] add script for netbird self hosted as lxc

### DISCUSSION THREAD

**Discussion by protonaut** [community-scripts/ProxmoxVE#644] [Request script]
### Application Name

netbird

### Website

https://docs.netbird.io/selfhosted/selfhosted-quickstart

### Description

As I recovered that netbird is quite smoothier and more polished than headscale I think a lot of people (including myself) would love to see a script for an netbird self-hosted server.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by Mati-l33t** [comment]
Is this not the same or what?
https://community-scripts.github.io/ProxmoxVE/scripts?id=add-netbird-lxc

**Reply by HammyHavoc** [reply]
That's for the NetBird client. You are mistaking it for https://docs.netbird.io/selfhosted/selfhosted-quickstart

**Comment by protonaut** [comment]
Thanks for the response. There is a script for the netbird-client already.
But this requires to login in the netbird-servers and therefore to register
on the (limited) conditions of  app.netbird.io.

What I am asking is a script for also hosting the netbird-server, so that
all the clients established with
https://community-scripts.github.io/ProxmoxVE/scripts?id=add-netbird-lxc

can log in to.

Am Mo., 2. Dez. 2024 um 19:24 Uhr schrieb Mati ***@***.***>:

**Reply by jimcuk** [reply]
Yes I would like to see this as well

**Comment by KrzysztofMaciejewski** [comment]
+1 for Netbird Server as LXC in Helper-Scripts

Thanks!

**Comment by svh1985** [comment]
+1 
https://github.com/community-scripts/ProxmoxVE/issues/6026

**Comment by MattXcz** [comment]
+1 to this one.... 
This was originaly requested on Dec 2, 2024... can we maybe move a little here ? 
From netbird installation guide, it is really simple now:
```
export NETBIRD_DOMAIN=netbird.example.com
curl -fsSL https://github.com/netbirdio/netbird/releases/latest/download/getting-started.sh | bash
```

**Comment by albotroz** [comment]
Excuse me. Just a question of understanding. Do I understand it right?
This here is only a Netbird Client? https://community-scripts.github.io/ProxmoxVE/scripts?id=add-netbird-lxc

Do you have to host (or buy/rent from Netbird) a Netbird server as well?

**Reply by Pyromane** [reply]
Netbird consists of the clients (there is already a script for this) and a central management instance.
You can use the instance hosted by Netbird (there is a free version with restrictions) or host the management instance yourself (this script request is for this purpose).

**Comment by marian-hummel** [comment]
+1

**Comment by saikhurana98** [comment]
+1

**Comment by MustThinkOfAName** [comment]
+1

**Reply by MickLesk** [reply]
https://community-scripts.org/scripts/netbird ??

**Reply by MustThinkOfAName** [reply]
There are two parts to netbird; a client and a server. I believe that what we have is the client. People are hoping we can get the server part.
https://netbird.io/knowledge-hub/proxmox-getting-started-guide
vs
https://docs.netbird.io/selfhosted/selfhosted-quickstart

**Reply by Pyromane** [reply]
The explanation of MustThinkOfAName  really hits the nail on the head

**Reply by MickLesk** [reply]
That is the Server Not the Client? The clients are the addon Scripts.

**Reply by MustThinkOfAName** [reply]
If you look at the current script on community-script, it installs the package `netbird-ui`.

If you look here: https://docs.netbird.io/get-started/install/linux 

It states 

> Linux Installation
> 
> The NetBird client (agent) allows a peer to join a pre-existing NetBird deployment. If a NetBird deployment is not yet available, there are both managed and [self-hosted](https://docs.netbird.io/selfhosted/selfhosted-quickstart) options available.

while installing netbird/netbird-ui (I'm not sure why community-script uses netbird-ui tbh)

**Reply by MickLesk** [reply]
The Script is from @TechHutTV himself. I never used anything of netbird. After reading the PR and everything that goes with it, my understanding is that it’s the server and that the other Netbird is the associated add-on

**Reply by MickLesk** [reply]
Having looked at the link and the accompanying script, I’m wondering why anyone would want to do this as an LXC? In other words, I would build it, but only using the bash call they provide, as the package only has a few branches and variants. In other words, it’s a plain Debian LXC with the executed Script inside?

**Reply by HammyHavoc** [reply]
> Having looked at the link and the accompanying script, I’m wondering why anyone would want to do this as an LXC? In other words, I would build it, but only using the bash call they provide, as the package only has a few branches and variants. In other words, it’s a plain Debian LXC with the executed Script inside?

If not an LXC, then what would you suggest as the ideal way to run a NetBird server? Not a trick question.

I've been running a NetBird server and NetBird "signal" from Docker containers for several years. I've mostly dropped Docker at this point because Proxmox makes way more sense for my needs. I can't imagine why I wouldn't want to run it in an LXC.

**Reply by TechHutTV** [reply]
If you're trying to run our server on Proxmox I'd recommend a Linux VM and follow the instructions here. https://docs.netbird.io/selfhosted/selfhosted-quickstart

I could work on a VM community script for this but it's pretty easy to set up this way. The script I made is just the client for a routing peer. We fixed an old DSN issue a few versions ago so you should be able to install NetBird client directly on host now.

**Reply by HammyHavoc** [reply]
> If you're trying to run our server on Proxmox I'd recommend up a Linux VM and follow the instructions here. https://docs.netbird.io/selfhosted/selfhosted-quickstart
> 
> I could work on a VM community script for this but it's pretty easy to set up this way. 
> 
> 

Why a VM and not an LXC?

**Reply by MickLesk** [reply]
Allright, thx for explanation. 

Then i take a look in next days. But i need some testers for this. Be patience.

**Reply by HammyHavoc** [reply]
> Allright, thx for explanation. 
> 
> Then i take a look in next days. But i need some testers for this. Be patience.

Happy to help test as it would be nice to drop the few remaining Docker containers on Unraid I haven't got round to migrating over to Proxmox.
